### PR TITLE
fix: auto height limits one row difference

### LINKED ISSUE #19079: [Feature]: Keep a buffer of min 1 row between min/max height in auto-height with limits
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Summary

Keep a buffer of min 1 row between min/max height in auto-height with limits

### Why should this be worked on?

Keep a buffer of min 1 row between min/max height in auto-height with limits so that the min/max handles never collide

### PR DESCRIPTION
**Pull request by ankurrsinghal**
## Description

Issue:-
Currently, the min and max limits handle in Auto Height with Limits move with each other on collision, while we can change the max when they have the same values but we cannot change min because the max handle is on top of it so they move together if we want to decrease the max.

Solution:-
We added a one-row difference between the two. When you increase the min towards the max, the max will increase but with one row extra, similarly, the min will decrease with one row extra when the max is decreased towards the min.

Fixes #19079

Media
> A video or a GIF is preferred. when using Loom, don’t embed because it looks like it’s a GIF. instead, just link to the video

## Type of change

- New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

- Manual

### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

File: app/client/src/components/autoHeightOverlay/index.tsx [hunk 1]
```diff
@@ -219,13 +235,21 @@ const AutoHeightOverlay: React.FC<AutoHeightOverlayProps> = memo(
 
     function onMaxStop() {
```
Comment (riodeuno) [inline app/client/src/components/autoHeightOverlay/index.tsx:236]: @ankurrsinghal Let us avoid defining functions within a component or another function. This is an anti-pattern and can cause performance issues.

**Review by riodeuno — CHANGES_REQUESTED** [review]
@ankurrsinghal A couple of requests to change. I haven't seen the logic in `autoHeightOverlay/index.tsx`. 

The `BaseWidget.tsx` changes look fine to me.

File: app/client/src/components/autoHeightOverlay/index.tsx (hunk 1 above)
Comment (ankurrsinghal) [reply app/client/src/components/autoHeightOverlay/index.tsx]: Wanted to do this for a long time, to slice up this file into smaller components and use useCallback for the child props. Made those changes.

File: app/client/src/components/autoHeightOverlay/index.tsx [hunk 2]
```diff
@@ -51,7 +51,7 @@ interface MinMaxHeightProps {
 interface AutoHeightOverlayContainerProps
   extends MinMaxHeightProps,
     WidgetProps {
-  batchUpdate: (height: number) => void;
+  batchUpdate: (options: { minHeight?: number; maxHeight?: number }) => void;
```
[the remark this answers is not in the document]
Comment (ankurrsinghal) [reply app/client/src/components/autoHeightOverlay/index.tsx:16]: Added!

**Review by riodeuno — APPROVED** [review]

**Comment by kamakshibhat-appsmith** [comment]
Looks Good  @ankurrsinghal

**Pull request merged**